### PR TITLE
Feature/forecast controllers test

### PR DESCRIPTION
**Pull request by rma-rripken**
Added integration test for forecast-spec and forecast-instance.
Also changed forecast-spec and forecast-instance to extend CwmsDTO and added fields to forecast-instance so that it could find its corresponding spec.

### CODE REVIEW

**Comment by rma-rripken** [comment]
I don't actually want the gradle file changed - I was just starting from devins pr https://github.com/USACE/cwms-data-api/pull/591

File: cwms-data-api/src/main/java/cwms/cda/data/dto/forecast/ForecastInstance.java [hunk 1]
```diff
@@ -1,130 +1,283 @@
...
-  }
+@FormattableWith(contentType = Formats.JSONV2, formatter = JsonV2.class)
+@JsonDeserialize(builder = ForecastInstance.Builder.class)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
+public class ForecastInstance extends CwmsDTO {
+
+    // unique identifier: spec + issueDateTime
+    // to find spec we need officeId, specId and locationId
+    // we get officeId from CwmsDTO
```
Comment (rma-rripken) [inline cwms-data-api/src/main/java/cwms/cda/data/dto/forecast/ForecastInstance.java]: I'm not sure if this class should have officeId, specId, locationId   
or should it just have the entire ForecastSpec?
or does it actually not need any of those in the actual object and the Controller methods (mostly create and update) should just take extra parameters?

Comment (adamkorynta) [reply cwms-data-api/src/main/java/cwms/cda/data/dto/forecast/ForecastInstance.java]: My vote is to include the spec. That's how I'm doing it in the hec-nucleus models: https://bitbucket.hecdev.net/projects/SC/repos/hec-nucleus/pull-requests/114/overview

File: cwms-data-api/src/main/java/cwms/cda/data/dto/forecast/ForecastSpec.java [hunk 2]
```diff
@@ -1,88 +1,90 @@
...
-  }
-
-  public List<TimeSeriesIdentifierDescriptor> getTimeSeriesIds() {
-    return timeSeriesIds;
-  }
-
-  public void validate() throws FieldException {
-    //TODO
-  }
+@FormattableWith(contentType = Formats.JSONV2, formatter = JsonV2.class)
```
Comment (adamkorynta) [inline cwms-data-api/src/main/java/cwms/cda/data/dto/forecast/ForecastSpec.java:28]: @rma-bryson was asking if this was enough for the DTO to show up in the OpenAPI? or does it only read the controllers and pick up DTO's referenced in the controller's annotations?

File: cwms-data-api/src/main/java/cwms/cda/data/dto/forecast/ForecastInstance.java (hunk 1 above)
Comment (adamkorynta) [reply cwms-data-api/src/main/java/cwms/cda/data/dto/forecast/ForecastInstance.java]: office, spec id, and designator would be the unique identifiers. Still not sure if the location is included in that

File: cwms-data-api/src/main/java/cwms/cda/data/dto/forecast/ForecastSpec.java (hunk 2 above)
Comment (rma-rripken) [reply cwms-data-api/src/main/java/cwms/cda/data/dto/forecast/ForecastSpec.java:28]: I suspect it needs to be referenced but I don't know

**Review by adamkorynta — DISMISSED** [review]

File: cwms-data-api/src/main/java/cwms/cda/data/dto/forecast/ForecastInstance.java (hunk 1 above)
Comment (adamkorynta) [reply cwms-data-api/src/main/java/cwms/cda/data/dto/forecast/ForecastInstance.java]: confirmed, location id is not included in the unique constraint

**Review by adamkorynta — APPROVED** [review]

**Pull request merged**